src/components/app.cjsx

```diff
@@ -1,35 +1,140 @@
+        </span>
+      </div>
+
+  renderEditingBody: ->
+    return null unless @state.exerciseId
+    if ExerciseStore.isLoading()
+      <div>Loading...</div>
+    else
+      <Exercise displayVertically={@state.verticalDisplay} exerciseId={@state.exerciseId} />
+
+  saveExercise: ->
+    id = @state.exerciseId
+    validity = ExerciseStore.validate(id)
+    if not validity?.valid
+      alert(validity?.reason or 'Not a valid exercise')
+      return
+
+    if confirm('Are you sure you want to save?')
+      if ExerciseStore.isNew(id)
+        ExerciseStore.on 'created', @redirect
+        ExerciseActions.create(id, ExerciseStore.get(id))
+      else
+        ExerciseActions.save(id)
 
   render: ->
-    if @props.id then return <Exercise id={@props.id} />
-
-    <BS.Grid>
-      <BS.Row><BS.Col xs={6}>
-        <p>
-          <label htmlFor="exercise-id">Exercise ID:</label>
-          <input id="exercise-id" type="number" onChange={@update} />
-        </p>
-        <p>
-          <a href="/exercises/#{@state.exerciseId}" className="btn btn-primary">
-            Go to Exercise
-          </a>
-        </p>
-      </BS.Col><BS.Col xs={6}>
-        <p className="btn btn-success add-exercise">
-          <a href="/exercises/new">
-            <i className="fa fa-plus-circle" />
-            Add New Exercise
-          </a>
-        </p>
-      </BS.Col></BS.Row>
-    </BS.Grid>
+    id = @state.exerciseId
+    classes = classnames('exercise', 'openstax', 'container-fluid',
+      {'is-loading': ExerciseStore.isLoading()}
+    )
+
+    isWorking = ExerciseStore.isSaving(id) or ExerciseStore.isPublishing(id)
+
+    <div className={classes}>
+      <ErrorModal />
+      <nav className="navbar navbar-default">
+        <div className="container-fluid">
+          <div className="navbar-header">
+            <BS.ButtonToolbar className="navbar-btn">
+              <a href="/exercises/new" className="btn btn-success">New Blank Exercise</a>
+              <AsyncButton
+                bsStyle='info'
+                onClick={@saveExercise}
+                disabled={isWorking}
+                isWaiting={ExerciseStore.isSaving(id)}
+                waitingText='Saving...'
+                isFailed={ExerciseStore.isFailed(id)}
+                >
+                Save Draft
+              </AsyncButton>
+              { if not ExerciseStore.isNew(id)
+                <AsyncButton
+                  bsStyle='primary'
+                  onClick={@publishExercise}
+                  disabled={not id or isWorking or ExerciseStore.isPublished(id)}
+                  isWaiting={ExerciseStore.isPublishing(id)}
+                  waitingText='Publishing...'
+                  isFailed={ExerciseStore.isFailed(id)}
+                  >
+                  Publish
+                </AsyncButton>
+              }
+              <form className="navbar-form navbar-right">
+                <BS.Input type="checkbox" label=" Split Vertically"
```

Comment:
Ah, why did you opt for a checkbox instead of responsive (via Bootstrap's grid)?